### ISSUE TITLE
Mark videos as "watched" for specified time frame

### ISSUE THREAD

**Issue by orthand** [FreeTubeApp/FreeTube#1088]
**Is your feature request related to a usage problem (not a bug)? Please describe.**
Usage problem additional to #357
When starting fresh with FreeTube or finding a new channel all older videos from newly subscribed channels will flood the subscription section.
Two possible scenarios where this is inconvenient:
- I've already watched most of these older videos and/or
- I won't catch up on all the older videos and am only interested in newer content

which is why I'm very hesitant to subscribe to new channels right now.

**Describe the solution you'd like to see implemented**
Add the possibility to mark videos as watched by their upload dates for a specified time frame/older than XY
- via a pop-up menu (embedded in the channel view (/additionally when subscribing))
- or by using a select option with something like CTRL+LMouse and +Shift for ranges (which could also later be used for functions like playlists).

**Describe alternatives you've considered**
In "Subscription" only show uploads from newly subscribed channels with upload dates after they have been added. Though I'm not the biggest fan of that myself and not sure if that would interfere/create unnecessary confusion with the "Hide Videos on Watch" setting (i.e. when accidentally unsubscribing or wanting to add older videos to the Subscription page by removing them from history). I was pretty sure there was an issue already discussing this alternative but for the life of me I can't find it anymore.

**Comment by GilgusMaximus** [comment]
Hi,
so the original reasoning seems a bit specific to be honest and I do not see the work time to time gain balance here. You can subscribe to a channel and then simply add the videos on the first page of subscriptions to the watch history via the three dots button and then all new videos will slowly replace the old ones (Maximum of 2 minutes). Additionally if you subscribe to enough channels that post regularly this issue sorts itself out very quickly with new videos rolling in.

However a mode to easily add many videos to the watch history at once would be a more general solution to this. Something like a selection mode where you simply click on the video elements on a page (channel, subscription ....) and it adds them automatically without opening the actual video. 

Best regards,
Gilgus

**Comment by orthand** [comment]
> However a mode to easily add many videos to the watch history at once would be a more general solution to this. Something like a selection mode where you simply click on the video elements on a page (channel, subscription ....) and it adds them automatically without opening the actual video.

Hi Gilgus, something like the selection mode is exactly what I was going for! (Another symbol on the thumbnail under playlist/fav would get too cluttered imo even as appear-on-hover. I used to hit the "watch later" on YT too often.)

And I just realized it's only adding the last 30 videos. I saw a few older than two years and thought it went wayyy further back so I get your confusion what this is all about. In case it's too specific feel free to close.

**Comment by GilgusMaximus** [comment]
Hi,
no problem. First might be a bit weird that such old videos are displayed. If the channel did not upload many videos that may certainly be the case. However I'd leave the issue open and just rephrase it to such a selection mode :)

**Comment by zeronobody0** [comment]
I do not know if this comment will be of much use, but I felt the need to mention that I second this request. Please do this.
